library/std/src/thread/mod.rs

```diff
@@ -1850,7 +1850,7 @@ impl<T> JoinHandle<T> {
     /// `is_finished`, and calling `join` if it returns `true`. This function does not block. To
     /// block while waiting on the thread to finish, use [`join`][Self::join].
     ///
-    /// This might return `true` for a brief moment after the thread's main
+    /// This might return `false` for a brief moment after the thread's main
```

Comment:
The original text (might return *true*) is what I intended here.

The point is that this might return true (finished) while the thread is technically still running, because it only checks if the main function has returned, not if the thread is actually gone (from the kernel's/system's perspective).

There's probably a clearer way to word that though. ^^'

Reply:
I see. What I had assumed with this PR is that `is_finished` may return "not finished" after the main function has returned but before some internal mechanism is notified of that.

Maybe both are true? Since there is not direct synchronization, if called close in time to the end of the thread `is_finished` is not super reliable either way but can be "good enough".